### PR TITLE
feat: Add support for http proxies with basic authentication

### PR DESCRIPTION
**Pull request by dennis-menge**
I am not sure about all the reflection stuff in the test.

### CODE REVIEW

**Comment by dennis-menge** [comment]
Also one question regarding code style/formatting, do you have anything defined there?

**Comment by ivarconr** [comment]
> Also one question regarding code style/formatting, do you have anything defined there?

It seems like you changed the file from "4 spaces indentation" in to "2 space indentation" making it a bit hard to see what you actually changed. 

Could you mind fix that?

**Comment by dennis-menge** [comment]
Should be better now I hope :)

File: src/main/java/no/finn/unleash/util/UnleashConfig.java [hunk 1]
```diff
@@ -45,32 +44,36 @@ public UnleashConfig(
...
             UnleashContextProvider contextProvider,
+            boolean isProxyAuthenticationByJvmProperties,
             boolean synchronousFetchOnInitialisation,
             UnleashScheduledExecutor unleashScheduledExecutor,
             UnleashSubscriber unleashSubscriber
     ) {
 
-
-        if(appName == null) {
+        if (appName == null) {
```
Comment (ivarconr) [inline src/main/java/no/finn/unleash/util/UnleashConfig.java]: no space between if and parenthesis.

File: src/main/java/no/finn/unleash/util/UnleashConfig.java [hunk 2]
```diff
@@ -45,32 +44,36 @@ public UnleashConfig(
...
             throw new IllegalStateException("You are required to specify the unleash appName");
         }
 
-        if(instanceId == null) {
+        if (instanceId == null) {
```
Comment (ivarconr) [inline src/main/java/no/finn/unleash/util/UnleashConfig.java]: no space between if and parenthesis.

File: src/main/java/no/finn/unleash/util/UnleashConfig.java [hunk 3]
```diff
@@ -45,32 +44,36 @@ public UnleashConfig(
...
             throw new IllegalStateException("You are required to specify the unleash instanceId");
         }
 
-        if(unleashAPI == null) {
+        if (unleashAPI == null) {
```
Comment (ivarconr) [inline src/main/java/no/finn/unleash/util/UnleashConfig.java]: no space between if and parenthesis.

File: src/main/java/no/finn/unleash/util/UnleashConfig.java [hunk 4]
```diff
@@ -45,32 +44,36 @@ public UnleashConfig(
...
             throw new IllegalStateException("You are required to specify the unleashAPI url");
         }
 
-        if(unleashScheduledExecutor == null) {
+        if (unleashScheduledExecutor == null) {
```
Comment (ivarconr) [inline src/main/java/no/finn/unleash/util/UnleashConfig.java]: no space between if and parenthesis.

File: src/main/java/no/finn/unleash/util/UnleashConfig.java [hunk 5]
```diff
@@ -45,32 +44,36 @@ public UnleashConfig(
...
             throw new IllegalStateException("You are required to specify a scheduler");
         }
 
-        if(unleashSubscriber == null) {
+        if (unleashSubscriber == null) {
```
Comment (ivarconr) [inline src/main/java/no/finn/unleash/util/UnleashConfig.java]: no space between if and parenthesis.

File: src/main/java/no/finn/unleash/util/UnleashConfig.java [hunk 6]
```diff
@@ -45,32 +44,36 @@ public UnleashConfig(
...
             throw new IllegalStateException("You are required to specify a subscriber");
         }
 
+        if (isProxyAuthenticationByJvmProperties) {
```
Comment (ivarconr) [inline src/main/java/no/finn/unleash/util/UnleashConfig.java]: no space between if and parenthesis.

**Comment by dennis-menge** [comment]
Hi @ivarconr I removed those blanks in UnleashConfig

**Comment by ivarconr** [comment]
Thanks for your contribution. Released as part of `3.2.10`.

**Pull request merged**